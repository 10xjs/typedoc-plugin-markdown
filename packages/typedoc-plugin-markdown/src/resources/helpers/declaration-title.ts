import { DeclarationReflection, ParameterReflection } from 'typedoc';
import { IntrinsicType } from 'typedoc/dist/lib/models';
import { memberSymbol } from './member-symbol';
import { type } from './type';

export function declarationTitle(
  this: ParameterReflection | DeclarationReflection,
  expandType = true,
) {
  const md = [memberSymbol.call(this) + ' '];

  if (this.flags && !this.flags.isRest) {
    md.push(this.flags.map((flag) => `\`${flag}\``).join(' '));
  }

  md.push(`${this.flags.isRest ? '... ' : ' '}**${this.name}**`);

  const type = getType(this, expandType);
  if (type) {
    md.push(`: ${type}`);
  }
  if (this.defaultValue) {
    md.push(` = ${this.defaultValue}`);
  }
  return md.join('');
}

function getType(
  reflection: ParameterReflection | DeclarationReflection,
  expandType: boolean,
) {
  if (
    reflection.type instanceof IntrinsicType &&
    reflection.type.name === 'object'
  ) {
    return type.call(reflection, expandType);
  }
  if (reflection instanceof DeclarationReflection && reflection.signatures) {
    return type.call(reflection, expandType);
  }
  return type.call(reflection.type, expandType);
}
