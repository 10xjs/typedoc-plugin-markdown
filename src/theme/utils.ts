import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
