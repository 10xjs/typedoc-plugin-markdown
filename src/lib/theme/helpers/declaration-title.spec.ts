import { Fixture, getFixture, getReflectionByName } from '../../../test/test.utils';
import { declarationTitle } from './declaration-title';

describe(`declarationTitle helper`, () => {
  test(`should compile`, () => {
    const data = getReflectionByName(getFixture(Fixture.Variable), 'myConst');
    const result = declarationTitle.call(data);
    expect(result).toMatchSnapshot();
  });
});
