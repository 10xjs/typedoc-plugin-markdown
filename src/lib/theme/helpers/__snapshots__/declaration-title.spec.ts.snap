// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declarationTitle helper should compile 1`] = `"**myConst**: *\`[object Object]\`* = 15"`;
